sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/join/CoGbkResult.java

```diff
@@ -98,8 +104,7 @@ public CoGbkResult(
         throw new IllegalStateException(
             "union tag " + unionTag + " has no corresponding tuple tag in the result schema");
       }
-      List<Object> valueList = (List<Object>) valueMap.get(unionTag);
-      valueList.add(value.getValue());
+      valuesByTag.get(unionTag).add(value.getValue());
     }
 
     if (taggedIter.hasNext()) {
```

Comment:
nit: use guard style

```
if (!taggedIter.hasNext()) {
  valueMap = (List) valuesByTag;
}

// If we get here, there were more elements than we can afford to
// keep in memory, so we copy the re-iterable of remaining items
// and append filtered views to each of the sorted lists computed earlier.
...
```

Reply:
There are pros and cons to this, but I agree it's a slight improvement. Done.